### PR TITLE
feat: add mdadm support

### PR DESCRIPTION
**Pull request by sergelogvinov**
Activate healthy mdraid at boot time.

Signed-off-by: Serge Logvinov <serge.logvinov@sinextra.dev>

### CODE REVIEW

File: internal/eudev/pkg.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: eudev
```
Comment (smira) [inline internal/eudev/pkg.yaml:1]: I feel we need a better story here, I don't want to duplicate every pkg we have in `pkgs/` already. I will think what we can do to provide better story here

Comment (frezbo) [reply internal/eudev/pkg.yaml:1]: I guess we can depend on the published image from `pkgs`

Comment (smira) [reply internal/eudev/pkg.yaml:1]: yep, exactly, I just wanted to to a bit of refactoring to make this easier

**Comment by dm3ch** [comment]
Am I right understood that this PR would allow having a root partition on software raid (mirror)?

By the way, I see that there were no activity for almost a year, so what is the actual status/plans? Maybe some help from outside needed?

**Comment by Gottox** [comment]
What's currently blocking? - I think I could make use of this extension ^^

**Comment by sergelogvinov** [comment]
Hello, I think this extension can be used only for csi-drivers (for non system disk)
For system disk we need to add support for boot loader and add some (huge) code in Talos.

Probably you can create raid manually and `dd` image to the soft raid after. But i did not try it.
I believe that mdadm binary should be a part of the system as lvm utils (which already there).

PS. only 2 like in this pr, so is not so popular...

**Comment by sergelogvinov** [comment]
new implementation https://github.com/siderolabs/extensions/pull/271

**Comment by sergelogvinov** [comment]
https://github.com/siderolabs/extensions/pull/271 was merged

**Pull request closed without merging**